pkg/test/framework/components/istio/util.go

```diff
@@ -67,12 +67,43 @@ func waitForValidationWebhook(accessor *kube.Accessor, cfg Config) error {
 	}, retry.Timeout(time.Minute))
 }
 
-func getRemoteDiscoveryAddress(cfg Config, cluster kubeenv.Cluster) (net.TCPAddr, error) {
+func GetRemoteDiscoveryAddress(namespace string, cluster kubeenv.Cluster, useNodePort bool) (net.TCPAddr, error) {
 	// If running in KinD, MetalLB must be installed to enable LoadBalancer resources
-	svc, err := cluster.GetService(cfg.SystemNamespace, igwServiceName)
+	svc, err := cluster.GetService(namespace, igwServiceName)
 	if err != nil {
 		return net.TCPAddr{}, err
 	}
+	// if useNodePort is set, we look for the node port service. This is generally used on kind or k8s without a LB
```

Comment:
you may also want to have a VMGateway running as node port. Traffic into the cluster can enter via the VM gateway which is setup as auto passthrough.